### PR TITLE
[reflection] No more mono_*_raise_*

### PR DESCRIPTION
**Pull request by lambdageek**
No more `mono_error_raise_exception` and `mono_raise_exception` in `reflection.c`

### CODE REVIEW

File: mono/metadata/reflection.c [hunk 1]
```diff
@@ -9728,31 +9729,39 @@ mono_reflection_resolve_custom_attribute_data (MonoReflectionMethod *ref_method,
...
 	*named_args = namedargs;
+
 leave:
 	g_free (arginfo);
-	mono_error_raise_exception (&error);
+	return mono_error_ok (error);
+}
 
+void
+mono_reflection_resolve_custom_attribute_data (MonoReflectionMethod *ref_method, MonoReflectionAssembly *assembly, gpointer data, guint32 len, MonoArray **ctor_args, MonoArray **named_args)
```
Comment (ludovic-henry) [inline mono/metadata/reflection.c]: why not call it `ves_icall_...` ?

File: mono/metadata/reflection.c [hunk 2]
```diff
@@ -12327,14 +12336,25 @@ mono_reflection_bind_generic_method_parameters (MonoReflectionMethod *rmethod, M
...
 	
-	MonoReflectionMethod *ret = mono_method_get_object_checked (mono_object_domain (rmethod), inflated, NULL, &error);
-	mono_error_raise_exception (&error); /* FIXME don't raise here */
+	MonoReflectionMethod *ret = mono_method_get_object_checked (mono_object_domain (rmethod), inflated, NULL, error);
 	return ret;
 }
 
+MonoReflectionMethod*
+mono_reflection_bind_generic_method_parameters (MonoReflectionMethod *rmethod, MonoArray *types)
+{
```
Comment (ludovic-henry) [inline mono/metadata/reflection.c]: why not call it `ves_icall_...`?

**Comment by ludovic-henry** [comment]
LGTM, thank you! :)

**Comment by lambdageek** [comment]
Thanks for reviewing @ludovic-henry .  I'll rename the icalls with the right naming convention.

**Comment by lambdageek** [comment]
@monojenkins merge

**Pull request merged**